pkg/rc/replication_controller.go

```diff
@@ -443,8 +446,68 @@ func (rc *replicationController) removePods(rcFields fields.RC, current types.Po
 			txn, cancelFunc = rc.newAuditingTransaction(context.Background(), rcFields, txn.Nodes())
 		}
 
-		unscheduleFrom, ok := preferred.PopAny()
+		unscheduleFrom, ok := ineligible.PopAny()
 		if !ok {
+			if rcFields.Disabled {
+				// Long explanation warning:
+				//
+				// Background: an RC is typically disabled when it is
+				// part of a rolling update, in which pods are
+				// transferred from the "old" (disabled) RC onto a
+				// "new" (enabled) RC.  This is done by decreasing
+				// the replica count on the old RC, and increasing
+				// the replica count on the new RC. Decrementing the
+				// count on the old RC was, until recently, mostly
+				// just a symbolic act because the RC is disabled, so
+				// it wasn't taking any action in response to the
+				// count changes. This works because when the new RC
+				// schedules the pod on a node that the old RC
+				// scheduled, it implicitly destroys the old instance
+				// because it overwrites the same consul key.
+				//
+				// With the introduction of node transfers, we can't
+				// quite rely on this behavior everywhere. Node
+				// transfers are meant to transfer a pod from an
+				// "ineligible" node to a new "eligible" node
+				// transparently to the rest of the system. One
+				// complication is that during a rolling update, a
+				// node on the "old" (disabled) RC might become
+				// ineligible. In this case, the "new" (enabled) RC
+				// is never going to know about the existence of this
+				// ineligible node because it has been removed from
+				// the scheduler, which means that the old RC is
+				// responsible for ultimately unscheduling the pod
+				// from this node.
+				//
+				// For this reason, we have this special case where a
+				// disabled RC is allowed to unschedule ineligible
+				// nodes, but it should NOT unschedule eligble nodes
+				// (because it will rely on a different enabled RC to
+				// do it implicitly)
+				//
+				// One final safety restriction we put in place
+				// is that a disabled RC should only unschedule
+				// ineligible nodes if its replica count is
+				// zero, which we expect to occur at the end of
+				// a rolling update. This protection is
+				// necessary to prevent the rolling update from
+				// counting this ineligible (presumably
```

Comment:
I don't agree that ineligible nodes will be healthy in all or even most cases. I don't think it's material for the logic tho.